File: app/src/main/java/kg/study/kbv_viewpager2/ImagePagerAdapter.java
package kg.study.kbv_viewpager2;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.Glide;
import com.flaviofaria.kenburnsview.KenBurnsView;
import com.squareup.picasso.Picasso;

import java.util.List;

public class ImagePagerAdapter extends RecyclerView.Adapter<ImagePagerAdapter.ImagePagerViewHolder> {

    private List<Image> imageList;

    public ImagePagerAdapter(List<Image> imageList) {
        this.imageList = imageList;
    }

    @NonNull
    @Override

    public ImagePagerViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new ImagePagerViewHolder(
                LayoutInflater.from(parent.getContext()).inflate(
                        R.layout.item_view_pager,
                        parent,
                        false
                )
        );
    }

    @Override
    public void onBindViewHolder(@NonNull ImagePagerViewHolder holder, int position) {
        holder.loadImage(imageList.get(position));
    }

    @Override
    public int getItemCount() {
        return imageList.size();
    }

    static class ImagePagerViewHolder extends RecyclerView.ViewHolder {

        private KenBurnsView kenBurnsView;

        ImagePagerViewHolder(@NonNull View itemView) {
            super(itemView);
            kenBurnsView = itemView.findViewById(R.id.ken_burns_view);

        }

        void loadImage(Image image) {
            Picasso.get().load(image.imageUrl).into(kenBurnsView);
        }
    }
}

File: app/src/main/java/kg/study/kbv_viewpager2/ImagePagerAdapter.kt
package kg.study.kbv_viewpager2

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import androidx.recyclerview.widget.RecyclerView.ViewHolder
import com.flaviofaria.kenburnsview.KenBurnsView
import com.squareup.picasso.Picasso
import kg.study.kbv_viewpager2.ImagePagerAdapter.ImagePagerViewHolder
import java.util.*

class ImagePagerAdapter : RecyclerView.Adapter<ImagePagerViewHolder>() {
    private var imageList: ArrayList<Image>? = null

    /* public ImagePagerAdapter(List<Image> imageList) {
        this.imageList = imageList;
    }*/
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ImagePagerViewHolder {
        return ImagePagerViewHolder(
                LayoutInflater.from(parent.context).inflate(
                        R.layout.item_view_pager,
                        parent,
                        false
                )
        )
    }

    override fun onBindViewHolder(holder: ImagePagerViewHolder, position: Int) {
        //Image currentImage = imageList.get(position);
        holder.loadImage(imageList!![position])
        //   holder.loadImage(setImageList(imageList));
    }

    override fun getItemCount(): Int {
        return imageList!!.size
    }

    class ImagePagerViewHolder(itemView: View) : ViewHolder(itemView) {
        private val kenBurnsView: KenBurnsView
        fun loadImage(image: Image) {
            Picasso.get().load(image.imageUrl).into(kenBurnsView)
        }

        init {
            kenBurnsView = itemView.findViewById(R.id.ken_burns_view)
        }
    }

    fun setImageList(images: ArrayList<Image>?) {
        imageList = images
        notifyDataSetChanged()
    }
}